src/envelope.rs

```diff
@@ -137,3 +139,52 @@ impl<A> Clone for Box<dyn BroadcastMessageEnvelope<Actor = A>> {
         BroadcastMessageEnvelope::clone(&**self)
     }
 }
+
+/// Like MessageEnvelope, but with an Arc instead of Box
+pub(crate) trait BroadcastEnvelope: HasPriority + Send + Sync {
+    type Actor;
+
+    fn handle<'a>(
+        self: Arc<Self>,
+        act: &'a mut Self::Actor,
+        ctx: &'a mut Context<Self::Actor>,
+    ) -> BoxFuture<'a, ()>;
+}
```

Comment:
That would be possible - I think it'd lead to a little more overhead for non-broadcast messages, though, which is the majority case, so I'm unsure if it's worth the simplification

Reply:
1) Initialization of two atomic integers (not sure if this actually costs anything - it might be free)
2) Extra data of two atomic integers (16 bytes extra per message)
3) Acquire fence in `Arc::drop`. Not sure if this is very costly.

Mostly, the extra data per message seems unnecessary to me, and probably worth the extra ~9 lines or so to avoid it (if I am counting correctly - it should just be the 9 defining the BroadcastEnvelope trait).

Reply:
I've managed to get rid of the `Arc` for broadcasting: https://github.com/Restioson/xtra/pull/86

Let me know what you think.

Reply:
Resolving this conversation in favour of discussing on #86